molo/core/utils.py

```diff
@@ -34,22 +34,30 @@ def copy_translation_pages(page, new_page):
             page=new_page,
             language=languages.filter(
                 is_main_language=True).first())
+        new_page.language = languages.filter(is_main_language=True).first()
 
-    for translation in page.translations.all():
-        new_lang = translation.translated_page.specific.copy_language(
+    for translation in page.specific.translated_pages.all():
+        new_lang = translation.specific.copy_language(
             current_site, destination_site)
-        new_translation = translation.translated_page.copy(
+        new_translation = translation.copy(
             to=new_page.get_parent())
         if LanguageRelation.objects.filter(page=new_translation).exists():
             new_l_rel = LanguageRelation.objects.get(page=new_translation)
             new_l_rel.language = new_lang
             new_l_rel.save()
+            new_translation.language = new_lang
         else:
             new_l_rel = LanguageRelation.objects.create(
                 page=new_translation, language=new_lang)
-        PageTranslation.objects.create(
-            page=new_page,
-            translated_page=new_translation)
+            new_translation.language = new_lang
+        for translated_page in \
+                page.specific.translated_pages.all():
```

Comment:
Why not add the new translations to the `translated_pages` for `new_page` as you create them? Then here you can loop through `new_page.translated_pages.all()` (ie loop through the new pages rather than the old ones)
Also, don't you need to add `new_page` to the other `translated_pages` at some point?

Reply:
@KaitCrawford it get's tricky because it complains about doing things inside the atomic block. This does add the new page to the other translated pages. I am working on adding the correct page for the new translated page. That happens in the copy method, this is just the copy translations method